prog/vnet/update_load_balancer_node.rb

```diff
@@ -129,22 +131,30 @@ def generate_lb_ip_set_definition(neighbor_vms)
       "elements = {#{neighbor_vms.map(&:private_ipv6).join(", ")}}"]
   end
 
-  def generate_lb_map_defs_ipv4(current_port)
-    load_balancer.active_vm_ports
+  def generate_lb_map_defs(current_port, address_selector)
+    items = load_balancer.active_vm_ports
       .select { |vm_port| vm_port.load_balancer_port.dst_port == current_port.dst_port }
-      .map
-      .with_index { |vm_port, index| "#{index} : #{vm_port.load_balancer_vm.vm.private_ipv4} . #{(vm_port.vm.id == vm.id) ? vm_port.load_balancer_port.dst_port : vm_port.load_balancer_port.src_port}" }
+      .map do |vm_port|
+        address = address_selector.call(vm_port)
+        port = (vm_port.load_balancer_vm.vm.id == vm.id) ? vm_port.load_balancer_port.dst_port : vm_port.load_balancer_port.src_port
+        [address.to_s, port.to_i]
+      end
+
+    items.sort_by { |address, port| [address, port] }.map.with_index do |(address, port), index|
+      "#{index} : #{address} . #{port}"
+    end
+  end
+
+  def generate_lb_map_defs_ipv4(current_port)
+    generate_lb_map_defs(current_port, ->(vm_port) {
+      vm_port.load_balancer_vm.vm.private_ipv4
+    })
```

Comment:
To pass a block:

```suggestion
    generate_lb_map_defs(current_port) do |vm_port|
      vm_port.load_balancer_vm.vm.private_ipv4
    end
```

You'll want to make a similar change for the other call.